services/web-server/src/login/strategies/github.js

```diff
@@ -82,8 +105,10 @@ module.exports = class Github {
           clientID: strategyCfg.clientId,
           clientSecret: strategyCfg.clientSecret,
           callbackURL: `${cfg.app.publicUrl}${callback}`,
+          scope: 'repo',
         },
         async (accessToken, refreshToken, profile, next) => {
+          this.githubClient = new GithubClient({ accessToken });
```

Comment:
This is setting a global variable (global to the strategy) based on credentials for a single user.  Since these are async operations that could overlap, that could lead to one user getting another user's credentials!  Let's not do that :)

This leads to a bigger problem: if we need the accessToken from the OAuth transaction to get the user's team memberships, then user scanning won't work.  It's cleverly hidden here by the global variable, but when the scanner runs, it will fail with something like `this.githubClient.listTeams is undefined`.

Solving that might be a bit of a puzzle.  @owlishDeveloper might be able to help!

Reply:
Those are valid points. We could use a `Map` to avoid this but I don't think this would solve the second point re: "scanner failing" since we won't have the user's oauth access token. One idea that comes to mind is to store the access token in an azure table keyed by the user's identity. This would allow the scanner to read public & private team memberships of a user and as far as I can tell, the oauth github token has no expiration. Does this make any sense? @owlishDeveloper @djmitche Did you have something else in mind?

Reply:
I will need to think about this a bit, and familiarize myself with the code. I'll get back to you asap

Reply:
Let me know if you had any questions or if there's anything that is not making sense. This whole thing is very confusing.

Reply:
owlish and I spoke on Slack and agreed to store the access token in a db.